scipy/_lib/_array_api.py

```diff
@@ -317,13 +322,15 @@ def xp_assert_equal(actual, desired, *, check_namespace=True, check_dtype=True,
         return xp.testing.assert_close(actual, desired, rtol=0, atol=0, equal_nan=True,
                                        check_dtype=False, msg=err_msg)
     # JAX uses `np.testing`
+    actual = _asarray(actual, xp_in=xp, xp_out=np)
+    desired = _asarray(desired, xp_in=xp, xp_out=np)
```

Comment:
Both the actual and desired arrays should of the same type already. If JAX and NumPy are the only array libraries that get here, and both of them work with `np.testing`, why is a conversion necessary?

Reply:
array-api-strict gets here too, and those arrays are incompatible with `np.testing` when `__array__` is removed.

Reply:
Ah, I see. Maybe we should do the conversion in the xp_asserts after checking that the namespaces match?

Reply:
Never mind. Just woke up.

Reply:
OK, I think the reason I asked originally is because if JAX uses `np.testing`, we wouldn't want to convert it to a NumPy array if it's on another device. So I think i'll change this to convert to NumPy only if it's array-api-strict?